kuebiko/pipeline/sic/flow.py

```diff
@@ -0,0 +1,146 @@
+from typing import Dict, List, Union
+
+from metaflow import FlowSpec, project, step
+
+## Declare constants in all caps at the top.
+URL = (
+    "http://www.ons.gov.uk/file?uri=/methodology/classificationsandstandards/"
+    "ukstandardindustrialclassificationofeconomicactivities/uksic2007/"
+    "sic2007summaryofstructurtcm6.xls"
+)
+
+# Type aliases
+field_name = str
+code_name = str
+code = str
+
+
+@project(name="kuebiko")
+class Sic2007Structure(FlowSpec):
+    """SIC 2007 taxonomy structure.
+
+    Includes extra subclasses used by companies house:
+    - 74990 - Non-trading company
+    - 98000 - Residents property management
+    - 99999 - Dormant company
+
+    Attributes:
+        url: Source excel file
+        records: Taxonomy structure as list of records.
+        section_lookup: Lookup from code to name
+        division_lookup: Lookup from code to name
+        group_lookup: Lookup from code to name
+        class_lookup: Lookup from code to name
+        subclass_lookup: Lookup from code to name
+    """
+
+    ## Type-hints for the data artifacts produced are super helpful!
+    ## Here we've used type aliases to make these easier to interpret.
+    url: str
+    records: List[Dict[field_name, Union[code, code_name]]]
+    section_lookup: Dict[code, code_name]
+    division_lookup: Dict[code, code_name]
+    group_lookup: Dict[code, code_name]
+    class_lookup: Dict[code, code_name]
+    subclass_lookup: Dict[code, code_name]
+
+    @step
+    def start(self):
+        """Fetch Excel spreadsheet containing taxonomy structure from ONS website."""
+        ## **Imports used with a step should always go within a step.**
+        ## Because metaflow has the ability to abstract infrastructure then
+        ## different steps may run on different machines with different environments.
+        from utils import get, excel_to_df
+
+        ## **Capture important values as data artifacts**, such as the URL data
+        ## was fetched from. This helps debugging previous runs and can monitor
+        ## important changes.<br>
+        ## Though be wary of doing this uneccessarily for large values due to
+        ## the overhead of persisting artifacts.
+        self.url = URL
+
+        ## Naming an artifact with a leading underscore makes the data artifact
+        ## available to future steps as usual but hides it from a user inspecting
+        ## the results of the run.
+        ## This is useful to avoid users loading the raw data rather than the processed
+        ## data, but it makes it harder to debug/monitor artifacts so it's a case
+        ## of best judgement.
+        self._raw_data = excel_to_df(get(self.url))
```

Comment:
I guess `self._raw_data` gets persisted as a data artifact of the run?

I would also instinctively want to persist somewhere the raw input .xlsx file, but I guess that's not necessary if we're keeping `self._raw_data`? (+ have faith that `excel_to_df` is faithfully transforming the input excel file)